charts/newrelic-infra-operator/templates/clusterrole.yaml

```diff
@@ -0,0 +1,43 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ template "newrelic-infra-operator.fullname" . }}
+  labels:
+    {{- include "newrelic-infra-operator.labels" . | nindent 4 }}
+rules:
+  - apiGroups: [""]
+    resources:
+      - "secrets"
+    verbs: ["get", "update", "patch"]
+    resourceNames: [{{ template "newrelic-infra-operator.fullname" . }}-config]
+  - apiGroups: [""]
+    resources:
+      - "secrets"
+    verbs: ["create"]
+  - apiGroups: ["rbac.authorization.k8s.io"]
+    resources: ["clusterrolebindings"]
+    verbs: ["list", "watch", "get"]
+  {{/* Our controller needs permission to add the ServiceAccounts from the user to the -infra-agent CRB */}}
+  - apiGroups: ["rbac.authorization.k8s.io"]
+    resources: ["clusterrolebindings"]
+    verbs: ["update"]
+    resourceNames: [{{ template "newrelic-infra-operator.fullname" . }}-infra-agent]
+---
+{{/* infra-agent is the ClusterRole to be used by the injected agents to get metrics */}}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ template "newrelic-infra-operator.fullname" . }}-infra-agent
+  labels: {{- include "newrelic-infra-operator.labels" . | nindent 4 }}
+rules:
+  - apiGroups: [""]
+    resources:
+      - "nodes"
+      - "nodes/metrics"
+      - "nodes/stats"
+      - "nodes/proxy"
+      - "pods"
+      - "services"
+    verbs: ["get", "list"]
+  - nonResourceURLs: ["/metrics"]
+    verbs: ["get"]
```

Comment:
Isn't this used for API monitoring which we are not doing right now?

Reply:
I believe this allows us to fetch `/metrics/summary` from Kubelet, which we are still doing (although through the API server proxy). I'm not sure if this is still required if we do it from there, maybe @paologallinaharbur can provide some light on this.